app/src/org/commcare/tasks/ResourceEngineTask.java

```diff
@@ -207,4 +216,28 @@ public void simpleResourceAdded() {
     public void incrementProgress(int complete, int total) {
         this.publishProgress(new int[]{complete, total, phase});
     }
+
+    public UnresolvedResourceException getMissingResourceException() {
+        return missingResourceException;
+    }
+
+    public InvalidResourceException getInvalidResourceException() {
+        return invalidResourceException;
+    }
+
+    public String getvAvailable() {

```

Comment:
Can these be renamed to `getVersionAvailable` and `getVersionRequired` ? Renaming the fields wouldn't be bad either.